src/check_missing_alleles.py

```diff
@@ -0,0 +1,439 @@
+    for allele in missing_alleles:
+        gene = allele.split("*")[0]
+        missing_genes.add(gene)
+
+    mro_chains = set()
+    mro_genes = set()
+    with open("../ontology/chain.tsv") as fh:
+        rows = csv.DictReader(fh, delimiter="\t")
+        for row in rows:
+            label = row["Label"]
+            if "HLA" in label:
+                if "*" not in label:
+                    gene = label.split(" ")[0][4:]
+                    mro_genes.add(gene)
+                else:
+                    allele = label.split(" ")[0][4:]
+                    mro_chains.add(allele)
+
+    new_genes = missing_genes.difference(mro_genes)
+    new_gene_tups = set()
+    for gene in new_genes:
+        label = f"HLA-{gene} chain"
+        synonyms = ""
+        class_type = "equivalent"
+        parent = "protein"
+        gene = f"HLA-{gene} locus"
+        new_gene_tups.add((label, synonyms, class_type, parent, gene))
+
+    new_chains = missing_alleles.difference(mro_chains)
+    new_chain_tups = set()
+    for allele in new_chains:
+        label = f"HLA-{allele} chain"
+        synonyms = ""
+        class_type = "subclass"
+        # This removes the specific allele # (generic chain)
+        parent = f"HLA-{allele.split('*')[0]} chain"
+        gene = ""
+        new_chain_tups.add((label, synonyms, class_type, parent, gene))
+
+    with open("../ontology/chain.tsv", "a+") as fh:
+        for tup in new_gene_tups:
+            fh.write(("\t").join(tup) + "\n")
+        for tup in new_chain_tups:
+            fh.write(("\t").join(tup) + "\n")
+
+    return new_genes
+
+
+def update_locus(missing_genes):
+    """Update genetic-locus.tsv with any missing genes
+
+    Returns:
+        A list of missing genetic loci ['DRB', 'Y']
+    """
+    class_I = ["A", "B", "C"]
+    class_II = ["DP", "DM", "DOA", "DOB", "DQ", "DR"]
+
+    mro_loci = set()
+    with open("../ontology/genetic-locus.tsv") as fh:
+        rows = csv.DictReader(fh, delimiter="\t")
+        for row in rows:
+            locus = row["Label"]
+            if "HLA" in locus:
+                mro_loci.add(locus.split(" ")[0][4:])
+
+    new_loci_rows = set()
+    new_loci = missing_genes.difference(mro_loci)
+    for locus in new_loci:
+        # Fast and dirty way to check if gene is classI, classII
+        mhc_class = ""
+        for prefix in class_I:
+            if prefix in locus:
+                mhc_class = "MHC class I"
+        for prefix in class_II:
+            if prefix in locus:
+                mhc_class = "MHC class II"
+        if mhc_class == "":
+            mhc_class = "non-classical MHC"
+
+        label = f"HLA-{locus} locus"
+        synonyms = ""
+        class_type = "subclass"
+        parent = f"human {mhc_class} locus"
+        new_loci_rows.add((label, synonyms, class_type, parent))
+
+    with open("../ontology/genetic-locus.tsv", "a+") as fh:
+        for row in new_loci_rows:
+            fh.write(("\t").join(row) + "\n")
```

Comment:
`csv.writer(fh, delimiter="\t", lineterminator="\n")` might be a better choice here